spec.bs

```diff
@@ -252,19 +252,216 @@ dictionary AuctionAdConfig {
+    1. If |k| [=string/starts with=] "browserSignals.", [=exception/throw=] a {{TypeError}}.
+    1. Set |signals|[|k|] to |v|.
+  1. If |key| equals to "*", then set |auctionConfig|'s
+    [=auction config/all buyers priority signals=] to |value|, and [=iteration/continue=].
+  1. Let |buyer| be the result of [=parsing an origin=] with |key|. If it fails, [=exception/throw=]
+    a {{TypeError}}.
+  1. [=map/Set=] |auctionConfig|'s [=auction config/per buyer priority signals=][|buyer|] to
+    |signals|.
+1. If |config|["{{AuctionAdConfig/sellerTimeout}}"] [=map/exists=], set |auctionConfig|'s
+  [=auction config/seller timeout=] to |config|["{{AuctionAdConfig/sellerTimeout}}"] milliseconds.
+1. [=list/For each=] |component| in |config|["{{AuctionAdConfig/componentAuctions}}"]:
+  1. If |isTopLevel| is false, [=exception/throw=] a {{TypeError}}.
+  1. Let |componentAuction| be the result of running [=validate and convert auction ad config=] with
+    |component| and [=validate and convert auction ad config/isTopLevel=] set to false.
+  1. [=list/Append=] |componentAuction| to |auctionConfig|'s [=auction config/component auctions=].
+1. If |config|["{{AuctionAdConfig/sellerExperimentGroupId}}"] [=map/exists=]:
+  1. Set |auctionConfig|'s [=auction config/seller experiment group id=] to
+    |config|["{{AuctionAdConfig/sellerExperimentGroupId}}"].
+1. Return |auctionConfig|.
+
+</div>
+
+To <dfn>parse an origin</dfn> given a [=string=] |input|:
+1. Let |url| be the result of running the [=URL parser=] on |input|.
+1. If |url| is an error, then return failure.
+1. Return |url|'s [=url/origin=].
+
+To <dfn>start bidding and scoring</dfn> given an [=auction config=] |auctionConfig|:
+1. If |auctionConfig|'s [=auction config/component auctions=] are empty:
+  1. TODO: Create seller worklet.
+1. OtherWise:
+  1. [=list/For each=] |component| in |auctionConfig|'s [=auction config/component auctions=]:
+    1. [=Start bidding and scoring=] with |component|.
+1. Create bidder worklets [=in parallel=]. [=list/For each=] |buyer| in
+  [=auction config/interest group buyers=]:
+  1. Let |bidderWorklet| be a [=bidder worklet=] object for |buyer|. TODO: expand.
+  1. Let |biddingScript| be the result of [=fetching a bidding javascript=] with |bidderWorklet|.
+  1. If |biddingScript| is an error, [=iteration/continue=].
+  1. TODO: Run all [=generate a bid|GenerateBid()=] tasks that are ready.
+  1. TODO: Maybe rewrite this part.
+
+To <dfn>create a request</dfn> given a [=url=] |url|, and a [=string=] <dfn for="create a request">
+|accept|</dfn>:
+1. Let |request| be a new [=request=] with the following properties:
+    :   [=request/URL=]
+    ::  |url|
+    :   [=request/header list=]
+    ::  A new [=header list=] containing a [=header=] named "`Accept`" whose value is |accept|
+    :   [=request/redirect mode=]
+    :: "`error`"
+    :   [=request/referrer=]
+    :: "`no-referrer`"
+    :   [=request/client=]
+    ::  `null`

```

Comment:
With no client, we also need to set the request's [origin](https://fetch.spec.whatwg.org/#concept-request-origin) (probably to an [=opaque origin=] for now, but maybe to the caller's origin) and [policy container](https://fetch.spec.whatwg.org/#concept-request-policy-container) (probably to a new [policy container](https://html.spec.whatwg.org/multipage/origin.html#policy-container)).